src/tarantool-gdb.py

```diff
@@ -0,0 +1,1371 @@
+        s += '['
+        if depth == 0:
+            s += '...]'
+            return
+        depth -= 1
+        if cls.typeof(data) != cls.MP_ARRAY:
+            return -1
+        arr_size = cls.decode_array(data)
+        for i in range(0, arr_size):
+            if i != 0:
+                s += ', '
+            s += cls.to_string_error_one(data, depth)
+        s += ']'
+
+    @classmethod
+    def to_string_error(cls, data, depth): # mp_print_error
+        s = str()
+        s += '{'
+        if depth == 0:
+            s += '...}'
+            return
+        depth -= 1
+        if cls.typeof(data) != cls.MP_MAP:
+            return -1
+        map_size = cls.decode_map(data)
+        for i in range(0, map_size):
+            if i != 0:
+                s += ', '
+            if cls.typeof(data) != cls.MP_UINT:
+                return -1
+            key = cls.decode_uint(data)
+            if key == cls.MP_ERROR_STACK:
+                s += '"stack": '
+                s += cls.to_string_error_stack(data, depth)
+            else:
+                s += '{}: '.format(key)
+                s += cls.to_string_data(data, depth)
+        s += '}'
+
+    @classmethod
+    def decode_compression(cls, data, len):
+        pos = data.pos()
+        self = Compression(dict(
+            type = cls.decode_uint(data),
+            raw_size = cls.decode_uint(data),
+            size = len - (data.pos() - pos),
+        ))
+        data.seek(self.size)
+        return self
+
+    @classmethod
+    def to_string_compression(cls, data, len): # mp_snprint_compression
+        ext = cls.decode_compression(data, len)
+        if ext is None:
+            return
+        return str(ext)
+
+    @classmethod
+    def decode_interval_field(cls, data):
+        mp_type = cls.typeof(data)
+        if mp_type == cls.MP_UINT:
+            value = cls.decode_uint(data)
+        elif mp_type == cls.MP_INT:
+            value = cls.decode_int(data)
+        else:
+            raise gdb.GdbError('Interval.decode_field: unexpected type ({})'.format(mp_type))
+        return value
+
+    INTERVAL_FIELD_YEAR = find_value('FIELD_YEAR')
+    INTERVAL_FIELD_MONTH = find_value('FIELD_MONTH')
+    INTERVAL_FIELD_WEEK = find_value('FIELD_WEEK')
+    INTERVAL_FIELD_DAY = find_value('FIELD_DAY')
+    INTERVAL_FIELD_HOUR = find_value('FIELD_HOUR')
+    INTERVAL_FIELD_MINUTE = find_value('FIELD_MINUTE')
+    INTERVAL_FIELD_SECOND = find_value('FIELD_SECOND')
+    INTERVAL_FIELD_NANOSECOND = find_value('FIELD_NANOSECOND')
+    INTERVAL_FIELD_ADJUST = find_value('FIELD_ADJUST')
+
+    DT_SNAP = find_value('DT_SNAP')
+
+    @classmethod
+    def decode_interval(cls, data, len): # interval_unpack
+        pos = data.pos()
+        itv = cls()
+        num_fields = data.read_u8()
+        for _ in range(0, num_fields):
+            field = data.read_u8()
+            value = cls.decode_interval_field(data)
+            if field == cls.INTERVAL_FIELD_YEAR:
```

Comment:
It's a replication of c-function `interval_unpack` and mapping switch operator into if-elif-else is used because it's simple and straightforward and visually closer to original switch, so I'd like to keep it to simplify synchronization if needed. Besides, it doesn't seem that we need performance optimization here.